snapshot-format.mediawiki

```diff
@@ -0,0 +1,76 @@
+up synchronization, lowers local storage needs, and ensures correctness via root-hash validation.
+Meanwhile, consensus nodes or archival nodes can continue to store the entire chain, ensuring full
+auditability and historical access for the network.
+
+==Specification==
+
+===MPT Snapshot Structure===
+
+The MPT state snapshot file '''MUST''' have the following layout, using little-endian encodings and standard
+varuint format for length fields:
+<ol class="nested-list">
+    <li>'''Height''' (4 bytes, <code>uint32</code>): The block index at which this state was generated.
+    <li>'''Root''' (32 bytes, <code>uint256</code> in LE): The state root hash corresponds to `Height`.
+    <li>'''NodeCount''' (<code>varuint</code>): The total number of MPT nodes in this snapshot.
+    <li>'''Nodes''': For each MPT node in the dump:
+        <ol>
+            <li>'''NodeLength''' (<code>varuint</code>): the size of the serialized node in bytes.
+            <li>'''NodeData''' (<code>NodeLength</code> bytes): the raw serialized MPT node.
+        </ol>
+    </li>
+</ol>
+
+A complete structure looks like this in memory:
+
+<code>
+| Height (4 bytes) | Root (32 bytes) | NodeCount (varuint) | [ NodeLength (varuint), NodeData... ] ... |
+</code>
+
+
+Implementations '''MUST''' write and read these fields in the exact order shown. No additional fields
+are required. Optional metadata (e.g., timestamps, attributes) '''MAY''' be written separately or
+stored alongside the file in a higher-level container system (e.g., object attributes in NeoFS).
+The '''Root''' field allows nodes to verify they’ve reconstructed the correct MPT.
+
+== Rationale ==
+
+Currently, the Neo node can synchronize only via P2P, going through the whole chain from genesis to the
+current block and storing all of the state locally. While P2P MPT state exchange can help, it still requires
+transferring large amounts of incremental data.
+Most nodes only need the most recent (or near-recent) state rather than the entire historical ledger.
+By standardizing an MPT-based snapshot format, nodes can skip fetching all previous blocks and simply
+load a validated state at a given height and then synchronize some small amount of blocks to catch up
+with other nodes. This not only reduces local storage consumption but also cuts down on bandwidth
+usage and synchronization times.
+
+In parallel, full-chain archival (either by consensus nodes or other designated roles) ensures that the
+complete history is still available for audits or specialized use cases. This strikes a balance between
+efficiency (for nodes that only want the current state) and full data retention (for those who need an
+archival record).
+
+== References ==
+* [https://github.com/neo-project/neo/issues/3463 Neo Issue#3463]
+
+== Implementation ==
+* Go: https://github.com/nspcc-dev/neo-go/pull/3808/commits/939eb9689988bdc0f6f031c393f0b0c07260a1bb
```

Comment:
We either need an issue to PR or to the commit in master, don't reference the unmerged commits.